ledger-core/virtual/integration/delegationtoken_test.go

```diff
@@ -301,83 +301,115 @@ func TestDelegationToken_IsMessageFromVirtualLegitimate(t *testing.T) {
-
-	server.Init(ctx)
-
-	var (
-		class       = gen.UniqueGlobalRef()
-		outgoing    = server.BuildRandomOutgoingWithPulse()
-		executorRef = server.RandomGlobalWithPulse()
-		firstPulse  = server.GetPulse()
-	)
-
-	server.IncrementPulse(ctx)
-
-	approver := server.RandomGlobalWithPulse()
-	jetCoordinatorMock.
-		MeMock.Return(approver).
-		QueryRoleMock.Return([]reference.Global{approver}, nil)
-
-	delegationToken := server.DelegationToken(reference.NewRecordOf(class, outgoing.GetLocal()), executorRef, outgoing)
-
-	server.IncrementPulse(ctx)
-
-	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
-	typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
-		assert.Equal(t, class, response.Callee)
-		assert.Equal(t, class, response.ResponseDelegationSpec.Callee)
-		assert.Equal(t, approver, response.ResponseDelegationSpec.Approver)
-		assert.Equal(t, outgoing, response.ResponseDelegationSpec.Outgoing)
-		return false
-	})
-
-	statePl := payload.VStateReport{
-		Status:                      payload.Empty,
-		Object:                      class,
-		OrderedPendingCount:         1,
-		OrderedPendingEarliestPulse: firstPulse.PulseNumber,
-	}
-	server.SendMessage(ctx, utils.NewRequestWrapper(server.GetPulse().PulseNumber, &statePl).SetSender(approver).Finalize())
-	server.WaitActiveThenIdleConveyor()
-
-	pl := payload.VDelegatedCallRequest{
-		Callee:         class,
-		CallFlags:      payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
-		CallOutgoing:   outgoing,
-		DelegationSpec: delegationToken,
-	}
-	msg := utils.NewRequestWrapper(server.GetPulse().PulseNumber, &pl).SetSender(executorRef).Finalize()
-	server.SendMessage(ctx, msg)
+			logger := utils.InterceptLog(inslogger.FromContext(ctx), logHandler)
+			server.OverrideConveyorFactoryLogContext(inslogger.SetLogger(ctx, logger))
+
+			server.Init(ctx)
+
+			var (
+				class       = gen.UniqueGlobalRef()
+				outgoing    = server.BuildRandomOutgoingWithPulse()
+				executorRef = server.RandomGlobalWithPulse()
+				firstPulse  = server.GetPulse()
+
+				delegationToken                     payload.CallDelegationToken
+				expectedVDelegatedCallResponseCount int
+			)
+
+			server.IncrementPulse(ctx)
+
+			approver := server.RandomGlobalWithPulse()
+			jetCoordinatorMock.
+				MeMock.Return(approver).
+				QueryRoleMock.Return([]reference.Global{approver}, nil)
+
+			if test.haveCorrectDT {
+				delegationToken = server.DelegationToken(reference.NewRecordOf(class, outgoing.GetLocal()), executorRef, outgoing)
+			}
+
+			server.IncrementPulse(ctx)
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VDelegatedCallResponse.Set(func(response *payload.VDelegatedCallResponse) bool {
+				assert.Equal(t, class, response.Callee)
+				assert.Equal(t, class, response.ResponseDelegationSpec.Callee)
+				assert.Equal(t, approver, response.ResponseDelegationSpec.Approver)
```

Comment:
maybe create expectedToken var/test field, so it would be asser.Equal(t, expectedToken, response.ResponseDelegationSpec)?

Reply:
done